fendermint/app/src/metrics/tracing.rs

```diff
@@ -57,6 +55,17 @@ macro_rules! set_gauge {
     };
 }
 
+/// Set a gague to the maximum of its value and a field in an event.
+macro_rules! max_gauge {
```

Comment:
There's usually no point in filtering data, as it is easily done in Prometheus / Mimir / Grafana.

And it could be problematic as it becomes not just a pure stream of data as is, but a complex filtered thing. To interpret it, one would need to take this Rust impl into account, which complicates things a lot if you're just operating Fendermint without being acknowledged with its metric reporting code base.

For example, if it would be `cmp::min`, then after restart `gauge.get()` would return zero, and it would always be zero. 

Here it is `cmp::max`, so the problem with default value after restart is not so pronounced, but you cannot keep this invariant of "never set values lower than before. Imagine, after the Fendermint restart, the gauge value will become 0, and `max` will return true.

So metrics collector could observe eg the following values `1 2 3 (restart) 2 3 4 5 6`, so this invariant no longer holds, as this is an in-memory local-only storage, it can't know the data in Prometheus.

The usual rule of thumb is to submit values as raw as possible, as to never hide any anomalies, and aggregate or normalise on Prometheus side.

Reply:
To be clear it was never my intention to promise an invariant that this can never go down, only that this is the highest value received since the node was started.

Reply:
And yes, a `min` version wouldn't make sense with a metric such as this, and without an exception made for 0.